code/segment_tree/model/segment_tree.py

```diff
@@ -0,0 +1,104 @@
+import copy
+import math
+
+
+class SegmentTree(object):
+    def __init__(self, tree_type):
+        if tree_type not in ['max', 'min', 'sum']:
+            raise TypeError('Incorrect type of segment tree. Expected: max\\min\\sum')
+        self.type = tree_type
+
+    def _build_tree(self, vertex, left, right):
+        if left == right:
+            self.tree[vertex] = self.array[left]
+        else:
+            mid = (left + right) // 2
+            self._build_tree(vertex * 2, left, mid)
+            self._build_tree(vertex * 2 + 1, mid + 1, right)
+
+            left_child = self.tree[vertex * 2]
+            right_child = self.tree[vertex * 2 + 1]
+
+            if self.type == 'sum':
+                self.tree[vertex] = left_child + right_child
+            elif self.type == 'min':
+                self.tree[vertex] = min(left_child, right_child)
+            elif self.type == 'max':
+                self.tree[vertex] = max(left_child, right_child)
```

Comment:
maybe it should be better to use dict with keys 'sum' 'min' 'max' and values with your operations?